frb_codegen/src/utils/misc.rs

```diff
@@ -184,3 +185,55 @@ macro_rules! ir {
         $crate::ir!($($rest)*);
     }
 }
+
+pub fn read_rust_file(path: &Path) -> String {
+    let path = path.to_str().unwrap();
+    let (dir, module) = get_dir_and_mod(path);
+    cargo_expand(&dir, module, path)
+}
+
+fn get_dir_and_mod(path: &str) -> (String, Option<String>) {
+    #[cfg(windows)]
+    let path = &path.replace('\\', "/");
+    let src_index = path
+        .rfind("/src/")
+        .expect("src dir must exist in rust project");
+    let dir = &path[..src_index];
+    #[cfg(windows)]
+    let dir = dir.strip_prefix("//?/").unwrap_or(dir);
+    let module = &path[src_index + 5..];
```

Comment:
nit: the magical "5" may be improved: what about `const NEEDLE = "/src/"`, and here only use `NEEDLE.len()`

Reply:
good point. Changed that